lib/stdlib/src/features.erl

```diff
@@ -0,0 +1,308 @@
+    ErrorMap#{reason => io_lib:format("~p: ~p", [?MODULE, Reason])}.
+
+format_error({invalid_features, Features}) ->
+    Fmt = fun F([Ftr]) -> io_lib:fwrite("'~p'", [Ftr]);
+              F([Ftr1, Ftr2]) ->
+                  io_lib:fwrite("'~p' and '~p'", [Ftr1, Ftr2]);
+              F([Ftr| Ftrs]) ->
+                  io_lib:fwrite("'~p', ~s", [Ftr, F(Ftrs)])
+          end,
+    case Features of
+        [Ftr] ->
+            io_lib:fwrite("the feature ~s does not exist.", [Fmt([Ftr])]);
+        Ftrs ->
+            io_lib:fwrite("the features ~s do not exist.", [Fmt(Ftrs)])
+    end.
+
+%% Hold the state of which features are currently enabled.
+%% This is almost static, so we go for an almost permanent state,
+%% i.e., user persistent_term.
+init_features() ->
+    persistent_term:put(enabled_features, []),
+    persistent_term:put(reserved_words, []),
+    case init:get_argument('enable-feature') of
+        error ->
+            %% no features enabled
+            ok;
+        {ok, Ftrs} ->
+            %% FIXME Need to be paranoid about any failures here.
+            %% Only convert to an existing atom.  This will also catch
+            %% a too long atom.
+            F = fun(String) ->
+                        case catch list_to_existing_atom(String) of
+                            {'EXIT', _} -> false;
+                            Atom -> {is_valid_feature(Atom), Atom}
+                        end
+                end,
+            lists:foreach(fun enable_feature/1,
+                          lists:filtermap(F, lists:append(Ftrs)))
+    end,
+    ok.
+
+enable_feature(Feature) ->
+    ?VALID_FEATURE(Feature),
+
+    Features = persistent_term:get(enabled_features),
+    case lists:member(Feature, Features) of
+        true ->
+            %% already there, maybe raise an error
+            Features;
+        false ->
+            NewFeatures = [Feature| Features],
+            persistent_term:put(enabled_features, NewFeatures),
+            Res = persistent_term:get(reserved_words),
+            New = reserved_words(Feature),
+            persistent_term:put(reserved_words, New ++ Res),
+            NewFeatures
+    end.
+
+disable_feature(Feature) ->
+    ?VALID_FEATURE(Feature),
+
+    Features = persistent_term:get(enabled_features),
+    case lists:member(Feature, Features) of
+        true ->
+            NewFeatures = Features -- [Feature],
+            persistent_term:put(enabled_features, NewFeatures),
+            Res = persistent_term:get(reserved_words),
+            Rem = reserved_words(Feature),
+            persistent_term:put(reserved_words, Res -- Rem),
+            NewFeatures;
+        false ->
+            %% Not there, possibly raise an error
+            Features
+    end.
+
+enabled_features() ->
+    persistent_term:get(enabled_features).
+
+reserved_words() ->
+    persistent_term:get(reserved_words).
+
+%% Return features used by module or beam file
+features_used(Module) ->
+    FileName =
```

Comment:
```suggestion
    FileName = module_to_filname(Module),
```

Reply:
Where the implementation of  module_to_filname is as shown below. I think it makes the code easier to read. Your code was not horribly v-shaped (as some code is) and it might be a little subjective, so this is my opinion for what it is worth.